test/database.py

```diff
@@ -269,6 +270,27 @@ def test_011_database_exists(self):
         # Now the database should exist
         assert db.exists()
 
+    def test_012_add_multiple_scripts_simultaneously(self):
```

Comment:
nit: please add a note that this test is technically non-deterministic and could end up looking flakey if a bug is introduced, since it relies on a race condition between the processes. 

Just in case someone observes weird flakey behavior in the future!

Reply:
Added!